### PR TITLE
155971785 copy calendar

### PR DESCRIPTION
**Pull request by mmarkuss**
# Added
- Adding a new trip is now allowed only after an user has defined a calendar schedule for the first trip.

### CODE REVIEW

File: ote/src/cljs/ote/views/route/trips.cljs [hunk 1]
```diff
@@ -136,20 +138,25 @@
 (defn trips-list [e! route]
   (let [stop-sequence (::transit/stops route)
         stop-count (count stop-sequence)
-        trips (::transit/trips route)]
-
+        trips (::transit/trips route)
+        empty-calendar? (empty? (first (:ote.db.transit/service-calendars route)))]
```
Comment (K1ll3rF0x) [inline ote/src/cljs/ote/views/route/trips.cljs]: Use ::transit/service-calendars

File: ote/src/cljs/ote/views/route/trips.cljs [hunk 2]
```diff
@@ -136,20 +138,25 @@
...
     [:div.route-times
      [:table {:style {:text-align "center"}}
       [route-times-header stop-sequence]
       [:tbody
        (doall (map-indexed (partial trip-row e! stop-count) trips))]]
+
+     (when empty-calendar?
+       [:div {:style {:margin-left "25px" :margin-right "25px" :margin-top "10px"}}
```
Comment (K1ll3rF0x) [inline ote/src/cljs/ote/views/route/trips.cljs]: Another opinion; the help box looks better if you remove margin-left and margin-right.

File: ote/src/cljs/ote/views/route/trips.cljs [hunk 3]
```diff
@@ -136,20 +138,25 @@
...
-
+        trips (::transit/trips route)
+        empty-calendar? (empty? (first (:ote.db.transit/service-calendars route)))]
     [:div.route-times
      [:table {:style {:text-align "center"}}
       [route-times-header stop-sequence]
       [:tbody
        (doall (map-indexed (partial trip-row e! stop-count) trips))]]
+
+     (when empty-calendar?
```
Comment (K1ll3rF0x) [inline ote/src/cljs/ote/views/route/trips.cljs]: This is an opinion, but wouldn't it be better to keep the infobox always visible? For example, I forgot the instructions after I opened the calendar open and went back to trips editor.

File: ote/src/cljs/ote/views/route/trips.cljs (hunk 1 above)
Comment (mmarkuss) [reply ote/src/cljs/ote/views/route/trips.cljs]: Fixed.

File: ote/src/cljs/ote/views/route/trips.cljs (hunk 2 above)
Comment (mmarkuss) [reply ote/src/cljs/ote/views/route/trips.cljs]: Fixed.

File: ote/src/cljs/ote/views/route/trips.cljs (hunk 3 above)
Comment (mmarkuss) [reply ote/src/cljs/ote/views/route/trips.cljs]: I'll leave this as it is now based on UI specs but we can remove it later if needed.

**Pull request merged**